kinoml/workflows/chembl22_cnn.py

```diff
@@ -0,0 +1,188 @@
+                    
+                if all(outputs_train[0] == elem for elem in outputs_train):
+                    print(f'The model predicts all values at {outputs_train[0]} .')
+
+                loss = criterion(outputs_train, y_train)
+                loss.backward()
+                optimizer.step()
+
+            print('---Predict')
+            # Test trained model on the test set
+            outputs_test = torch.squeeze(model(x_test))
+
+            # Transform target vectors into numpy array for scores
+            y_pred = outputs_test.detach().numpy()
+            y_test = y_test.detach().numpy()
+
+            # Transform probability of belonging to one class into 0 or 1 value at threshold 0.5
+            y_pred =(y_pred>0.5) 
+
+            if(all(y_pred[0] == elem for elem in y_pred)):
+                print(f'The model predicts all values at {y_pred[0]}')
+
+            print(f'# of elements on the test set: {len(y_test)}, # zero counts on the test set: {(y_test==0).sum()} ')
+
+            # Metrics on each fold
+            acc_per_fold.append(accuracy_score(y_test, y_pred))
+            conf_matrix_per_fold.append(confusion_matrix(y_test, y_pred).ravel())
+
+        # Mean on all K folds
+        acc_mean = np.mean(acc_per_fold)
+        conf_matrix_mean = np.mean(np.array(conf_matrix_per_fold), axis=0)
+
+        # Std on all K folds
+        acc_std = np.std(acc_per_fold)
+        conf_matrix_std = np.std(np.array(conf_matrix_per_fold), axis=0)
+
+        print(f' Metrics of CV for kinase {kinase} ')
+        print(f' Accuracy : Mean : {round(acc_mean, 2)} and std : {round(acc_std, 2)} ')
+        print(f' TN : Mean : {round(conf_matrix_mean[0], 2)} and std : {round(conf_matrix_std[0], 2)} ')
+        print(f' FP : Mean : {round(conf_matrix_mean[1], 2)} and std : {round(conf_matrix_std[1], 2)} ')
+        print(f' FN : Mean : {round(conf_matrix_mean[2], 2)} and std : {round(conf_matrix_std[2], 2)} ')
+        print(f' TP : Mean : {round(conf_matrix_mean[3], 2)} and std : {round(conf_matrix_std[3], 2)} ')
+
+        # Zero Counts per kinase
+        zero_count = (y==0).sum()
+        print(f' Total nb: {len(y)} and nb of zero counts : {zero_count}')
+        zero_count_per_kinase.append(zero_count)
+
+        acc_mean_per_kinase.append(acc_mean)
+        acc_std_per_kinase.append(acc_std)
+
+        conf_matrix_mean_per_kinase.append(conf_matrix_mean)
+        conf_matrix_std_per_kinase.append(conf_matrix_std)
+
+
+    # Save results to a pickle table in folder cv_results
+    results_cv = pd.DataFrame(data=[acc_mean_per_kinase, acc_std_per_kinase,
+                                    conf_matrix_mean_per_kinase, conf_matrix_std_per_kinase,
+                                    zero_count_per_kinase])
+    results_cv.columns = [data.columns[:(-2-k)]]
+    results_cv.index = ['Acc Mean', 'Acc Std', 'Conf Mean', 'Conf Std', 'Zero Count']
+    results_cv = results_cv.to_pickle('CNN_class_results_cv.pkl')
```

Comment:
Outputs should be parameterized too.

Reply:
(You can always set a default keyword argument in the hypothetical `main` function).